PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp

```diff
@@ -133,7 +131,30 @@ INT_PTR CALLBACK ProjectPanel::run_dlgProc(UINT message, WPARAM wParam, LPARAM l
 
         case WM_CONTEXTMENU:
 			if (!_treeView.isDragging())
-				showContextMenu(GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam));
+			{
+				int xPos = GET_X_LPARAM(lParam);
+				int yPos = GET_Y_LPARAM(lParam);
+
+				// If the context menu is generated from the keyboard, then we
+				// should display it at the location of the current selection
+				if (xPos == -1 && yPos == -1)
+				{
+					HTREEITEM selectedItem = _treeView.getSelection();
+
+					if (selectedItem)
+					{
+						RECT selectedItemRect;
+						if (TreeView_GetItemRect(_treeView.getHSelf(), selectedItem, &selectedItemRect, TRUE))
+						{
+							showContextMenuFromMenuKey(selectedItem, selectedItemRect.left, selectedItemRect.top);

```

Comment:
Its working correctly now, good job @luisffranca !

My only suggestion would be to change the offset a bit so the apps menu doesn't entirely obscure the selected item:
showContextMenuFromMenuKey(selectedItem, (selectedItemRect.left+selectedItemRect.right)/2, (selectedItemRect.top+ selectedItemRect.bottom)/2);

Reply:
@DavidHansen-0x454d43 I've done the change you suggested, and it looks better now :) Thanks!